code/game/turfs/open/floor/catwalk_plating.dm

```diff
@@ -7,37 +7,30 @@
  */
 /turf/open/floor/catwalk_floor	//the base type, meant to look like a maintenance panel
 	icon = 'icons/turf/floors/catwalk_plating.dmi'
-	icon_state = "maint_below"
+	icon_state = "maint_above"
 	name = "catwalk floor"
 	desc = "Flooring that shows its contents underneath. Engineers love it!"
 	baseturfs = /turf/open/floor/plating
 	floor_tile = /obj/item/stack/tile/catwalk_tile
+	layer = CATWALK_LAYER
+	plane = GAME_PLANE
 	footstep = FOOTSTEP_CATWALK
 	overfloor_placed = TRUE
 	underfloor_accessibility = UNDERFLOOR_VISIBLE
 	var/covered = TRUE
-	var/above_state = "maint_above"	//Icon-state for the overlay
-
+	var/catwalk_type = "maint"
 
 /turf/open/floor/catwalk_floor/Initialize(mapload)
 	. = ..()
+	var/mutable_appearance/plating_underlay = mutable_appearance(icon, "[catwalk_type]_below", TURF_LAYER)
+	underlays += plating_underlay
 	update_icon(UPDATE_OVERLAYS)
 
-GLOBAL_LIST_EMPTY(catwalk_overlay_masterlist)	//Stores all the above_states for the different types of catwalk
-
 /turf/open/floor/catwalk_floor/update_overlays()
+	layer = covered ? CATWALK_LAYER : TURF_LAYER
+	plane = covered ? GAME_PLANE : FLOOR_PLANE
+	icon_state = covered ? "[catwalk_type]_above" : "[catwalk_type]_below"
```

Comment:
Why are you updating the icon state in update overlays?

Reply:
I wasn't aware there was a different proc to do this in.  The updates were happening here before so I kept the location the same.

Reply:
You should look at how update_appearance works. There's different procs for different types of updates, and you can request one of the other depending on the situation

Reply:
On another note, what was the actual issue here? the use of overlays for some reason?

Reply:
> You should look at how update_appearance works.

Sounds good and I will try this.

> On another note, what was the actual issue here? the use of overlays for some reason?

Overlays for lighting and catwalks were glitching through the layers/planes to the current z-level the player was in.  When `transparent_turf` element was attached to a turf, the layer would be set to the `OPENSPACE_LAYER` which resulted in it being drawn as a bottom z-level with shadows around it.  This was more noticeable with glass floors.